File: backend/src/main/java/com/pairingplanet/pairing_planet/dto/user/UpdateProfileRequestDto.java
package com.pairingplanet.pairing_planet.dto.user;

import com.pairingplanet.pairing_planet.domain.enums.Gender;
import com.pairingplanet.pairing_planet.domain.enums.MeasurementPreference;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;

import java.time.LocalDate;
import java.util.UUID;

public record UpdateProfileRequestDto(
        @Size(max = 50, message = "Username cannot exceed 50 characters")
        String username,
        UUID profileImagePublicId, // [수정] String profileImageUrl -> UUID profileImagePublicId
        Gender gender,
        LocalDate birthDate,
        UUID preferredDietaryId,
        Boolean marketingAgreed,
        String locale,  // 언어 설정: ko-KR, en-US
        String defaultCookingStyle,  // 기본 요리 스타일: ISO country code (e.g., "KR", "US") or "international"
        MeasurementPreference measurementPreference,  // 측정 단위 선호: METRIC, US, ORIGINAL

        @Size(max = 150, message = "Bio cannot exceed 150 characters")
        String bio,

        @Pattern(
                regexp = "^$|^(https?://)?(www\\.)?(youtube\\.com/(channel/|c/|user/|@)[\\w-]+|youtu\\.be/[\\w-]+)/?$",
                message = "Invalid YouTube URL format"
        )
        String youtubeUrl,

        @Pattern(
                regexp = "^$|^@?[a-zA-Z0-9._]{1,30}$|^(https?://)?(www\\.)?instagram\\.com/[a-zA-Z0-9._]{1,30}/?$",
                message = "Invalid Instagram handle or URL format"
        )
        String instagramHandle
) {}

File: frontend_mobile/lib/data/models/user/update_profile_request_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'update_profile_request_dto.g.dart';

@JsonSerializable()
class UpdateProfileRequestDto {
  final String? username;
  final String? profileImagePublicId;
  final String? gender;  // MALE, FEMALE, OTHER
  final String? birthDate;  // yyyy-MM-dd format
  final String? preferredDietaryId;
  final bool? marketingAgreed;
  final String? locale;  // ko-KR, en-US
  final String? defaultCookingStyle;  // ISO country code (e.g., "KR", "US")
  final String? bio;  // User bio/description (max 150 chars)
  final String? youtubeUrl;  // YouTube channel URL
  final String? instagramHandle;  // Instagram handle

  UpdateProfileRequestDto({
    this.username,
    this.profileImagePublicId,
    this.gender,
    this.birthDate,
    this.preferredDietaryId,
    this.marketingAgreed,
    this.locale,
    this.defaultCookingStyle,
    this.bio,
    this.youtubeUrl,
    this.instagramHandle,
  });

  factory UpdateProfileRequestDto.fromJson(Map<String, dynamic> json) =>
      _$UpdateProfileRequestDtoFromJson(json);
  Map<String, dynamic> toJson() => _$UpdateProfileRequestDtoToJson(this);
}
